snippet-extractor-output/snippets.yaml

```diff
@@ -12691,20 +12715,44 @@
 
     const id = $("#notificationId").val().toString();
 
-    const details = 
-      {
+
+    let details;
+
+    // Identifies whether the current mail item is in read or compose mode.
+
+    if (Office.context.mailbox.item.itemId != undefined) {
+      // Read mode.
+      details = {
+        type: Office.MailboxEnums.ItemNotificationMessageType.InsightMessage,
+        message: "This is an insight notification with id = " + id,
+        icon: "icon1",
+        actions: [
+          {
+            actionText: "Open insight",
+            actionType: Office.MailboxEnums.ActionType.ShowTaskPane,
+            commandId: "PG.HelpCommand.Read",
+            contextData: { a: "aValue", b: "bValue" }
+          }
+        ]
+      };
+    } else {
+      // Compose mode.
+      details = {
         type: Office.MailboxEnums.ItemNotificationMessageType.InsightMessage,
-        message: "This is an insight notification",
+        message: "This is an insight notification with id = " + id,
         icon: "icon1",
         actions: [
           {
             actionText: "Open insight",
             actionType: Office.MailboxEnums.ActionType.ShowTaskPane,
-            commandId: "msgComposeOpenPaneButton",
+            commandId: "PG.HelpCommand.Compose",
```

Comment:
Same comment here. Would it be easier and more readable to scope the `if...else` check to the `commandId` attribute?